### DISCUSSION TITLE
Allow to add models

### DISCUSSION THREAD

**Discussion by Gummibeer** [spatie/laravel-sitemap#367] [Ideas]
It's likely that you want to add models like blog posts, products or similar to your sitemap.

The spatie RSS package has a `Feedable` interface for such cases. How about a `Sitemapable` interface which defines a method `getSitemapltem()` in which you have to return an instance of a sitemap tag.

This would allow to do `$sitemap->add($post)` and with another small adjustment would even allow to do `$sitemap->add(Post::all())`.

I think that the last one would be the biggest improvement as you can add all your posts, products ... in one line of code.

This would for sure also allow to add other classes implementing this interface - for example a custom sheet class from später sheet package.

**Comment by freekmurze** [comment]
Feel free to PR this. 

This package might be another good candidate to make PHP 8 only, do a cleanup, and add features like this.

**Reply by Gummibeer** [reply]
#368 is open. 🙂
